clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -694,13 +694,12 @@ static void VisitAccessorWrapperHelper(CXXRecordDecl *Owner, RangeTy Range,
     if (Util::isSyclAccessorType(ItemTy))
       (void)std::initializer_list<int>{
           (handlers.handleSyclAccessorType(Item, ItemTy), 0)...};
-    else if (ItemTy->isStructureOrClassType()) {
+    else if (Util::isSyclStreamType(ItemTy))
+      (void)std::initializer_list<int>{
+          (handlers.handleSyclStreamType(Item, ItemTy), 0)...};
+    else if (ItemTy->isStructureOrClassType())
```

Comment:
This is not correct. Current implementation of stream type requires recursive visiting its fields because stream wraps several accessors. 
I.e. this change will fail on `check-sycl`. We have poor testing of stream class in FE tests, I already created a community tracker to track it.

Reply:
Thanks! I wasn't sure.  I'll add the handling, but I'd like for it to stay a top-level entity in this if/else tree.  Else we get the awkward depth search like before.